holidays/ical.py

```diff
@@ -0,0 +1,225 @@
+                f"Invalid language tag: '{language}'. Expected format follows "
+                "ISO 639-1 or ISO 639-2, e.g., 'en', 'en-US'. For more details, "
+                "refer to: https://www.loc.gov/standards/iso639-2/php/code_list.php."
+            )
+        return language
+
+    def _fold_line(self, line: str) -> str:
+        """
+        Fold long lines according to RFC 5545.
+
+        Content lines SHOULD NOT exceed 75 octets. If a line is too long,
+        it must be split into multiple lines, with each continuation line
+        starting with a space.
+
+        :param line:
+            The content line to be folded.
+
+        :return:
+            The folded content line.
+        """
+        if line.isascii():
+            # Simple split for ASCII: every (CONTENT_LINE_MAX_LENGTH - 1) chars,
+            # as first char of the next line is space
+            if len(line) > CONTENT_LINE_MAX_LENGTH:
+                return CONTENT_LINE_DELIMITER_WRAP.join(
+                    line[i : i + CONTENT_LINE_MAX_LENGTH - 1]
+                    for i in range(0, len(line), CONTENT_LINE_MAX_LENGTH - 1)
+                )
+
+        elif len(line.encode()) > CONTENT_LINE_MAX_LENGTH:
+            # Handle non-ASCII text while respecting byte length
+            parts = []
+            part_start = 0
+            part_len = 0
+            for i, char in enumerate(line):
+                char_byte_len = len(char.encode())
+                part_len += char_byte_len
+
+                if part_len > CONTENT_LINE_MAX_LENGTH:
+                    parts.append(line[part_start:i])
+                    part_start = i
+                    part_len = char_byte_len + 1  # line start with space
+
+            parts.append(line[part_start:])
+            return CONTENT_LINE_DELIMITER_WRAP.join(parts)
+
+        # Return as-is if it doesn't exceed the limit
+        return line
+
+    def _generate_event(self, dt: date, holiday_name: str, holiday_length: int = 1) -> list[str]:
+        """
+        Generate a single holiday event.
+
+        :param dt:
+            Holiday date.
+
+        :param holiday_name:
+            Holiday name.
+
+        :param holiday_length:
+            Holiday length in days, default to 1.
+
+        :return:
+            List of iCalendar format event lines.
+        """
+        # Escape special characters per RFC 5545.
+        # SEMICOLON is used as a delimiter in HolidayBase (HOLIDAY_NAME_DELIMITER = "; "),
+        # so a name with a semicolon gets split into two separate `VEVENT`s.
+        sanitized_holiday_name = (
+            holiday_name.replace("\\", "\\\\").replace(",", "\\,").replace(":", "\\:")
+        )
+        event_uid = f"{uuid.uuid4()}@{self.holidays_version}.holidays.local"
+        if self.show_language and self.language is None:
+            raise ValueError(
+                "LANGUAGE cannot be included in SUMMARY as language code isn't provided"
+            )
```

Comment:
Maybe it's better to move this check to `__init__`? And modify the message without mentioning "SUMMARY" and more clearly pointing that language code is missing in the provided Holidays object?